tests/temp-humid-test.toit

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2025 Toit contributors.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the tests/LICENSE file.
+
+import expect show *
+import sensors
+import sensors.providers
+
+NAME ::= "toitware/sensors/test/temperature-humidity"
+MAJOR ::= 1
+MINOR ::= 0
+
+class TemperatureHumiditySensor implements providers.HumiditySensor providers.TemperatureSensor:
+  humidity-read -> float:
+    return 499.0
+
+  temperature-read -> float:
+    return 42.0
+
+main:
+  sensor := TemperatureHumiditySensor
+  provider-temperature := providers.TemperatureProvider NAME
```

Comment:
It seems a little annoying that you need to have two providers. I imagine that often a single driver will want to be the provider for multiple sensor services.

Reply:
as discussed in person: I will try to refactor the code a bit more.